Sparkle/SUBinaryDeltaCreate.m

```diff
@@ -138,11 +220,25 @@ int createBinaryDelta(NSString *source, NSString *destination, NSString *patchFi
         }
 
         NSDictionary *info = infoForFile(ent);
+        if (!info) {
+            fprintf(stderr, "Failed to retrieve info for file %s", ent->fts_path);
+            return 1;
+        }
         originalTreeState[key] = info;
+        
+        if (aclExists(ent)) {
+            fprintf(stderr, "Diffing ACLs are not supported. Detected ACL in before-tree on file %s", ent->fts_path);
+            return 1;
+        }
     }
     fts_close(fts);
+    
+    NSString *beforeHash = hashOfTreeWithVersion(source, majorVersion);
```

Comment:
OK, I was assuming it'd be easy to have a separate list of permissions to apply.